File: 88-merge-sorted-array/88-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int f =0;
        int s = 0;
        int nums[] = new int[m+n];
        int k =0;
        while (f < m && s < n) {
            if (nums1[f] < nums2[s]) {
                nums[k++] = nums1[f++];
            } else {
                nums[k++] = nums2[s++];
            }
            
        }
        while(f<m) {
            nums[k++] = nums1[f++];
        }
        while(s<n) {
            nums[k++] = nums2[s++];
        }
        for (int i =0;i<m+n;i++) {
            nums1[i] = nums[i];
        }
    }
}

File: 88-merge-sorted-array/88-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int end = m + n;
        while(m!=0&&n!=0)
        {
            if(nums1[m-1]>=nums2[n-1])
                nums1[--end]=nums1[--m];
            else
                nums1[--end]=nums2[--n];
        }
        while(n!=0)
            nums1[--end]=nums2[--n];
    }
};